### ISSUE TITLE
no-unused-import does not recognize @inheritdoc reference

### ISSUE THREAD

**Issue by jmendiola222** [protofire/solhint#659]
# Description

If a contract is only referenced by @inheritdoc documentation tag, you need to import it, but then the `no-unused-import` fails.

# Example

`A.sol`
```sol
contract A {
   /** 
   * @notice my cool foo funtion 
   */
   function foo() virtual view { ... }
}
```

`B.sol`
```sol
import { A } from A;

contract B is A {
}
```

`C.sol`
```sol
import { A } from A; // ----------------------------------------> THIS FAILS WITH no-unused-import
import { B } from B;

contract C is B {
    /**
     * @inheritdoc A -------------------------------------------> THIS REQUIRES "A" to be imported
     */
    function foo() override view { ... }
}
```

**Version**: 5.0.5

**Comment by dbale-altoros** [comment]
oh this is a good observation @jmendiola222 
thanks for that 
i will address it as soon as i can

**Comment by dbale-altoros** [comment]
@jmendiola222 
I'm testing your issue
And the rule is prepared for this kind of import and it doesn't give any false positive

`  it.only('should not raise', () => {
    const code = `
      import {A} from "./A.sol";
      import {B} from "./B.sol";

      contract C is B {
        /// @inheritdoc A
        function foo() external pure override {
          
        }
      }
      `

    const report = linter.processStr(code, {
      rules: { 'no-unused-import': 'error' },
    })
    assertNoErrors(report)
  })`

You can check in the `test/rules/best-practices/no-unused-import.js`
Last two tests are exactly the issue you are mentioning and it is not failing
